src/main/java/org/prebid/server/bidder/huaweiads/CountryCodeResolver.java

```diff
@@ -0,0 +1,362 @@
+                .filter(StringUtils::isNotBlank)
+                //mcc-mnc format
+                //todo: what if out of bound
+                .map(mccmnc -> mccmnc.split("-")[0])
+                .flatMap(CountryCodeResolver::countryFromMcc);
+
+        //todo: logic is slighty enhanced comparing to the PBS Go,
+        // if a country code of the device is invalid (has less than 3 symbols) I try to get the country from the user
+        // if the country of the user is invalid (has less than 3 symbols) I try to get the country from the MCC
+        // PBS Go returns default country code instead without trying to try to resolve code further
+        return countryOfDevice
+                .or(() -> countryOfUser)
+                .or(() -> countryOfMcc);
+    }
+
+    // convertCountryCode: ISO 3166-1 Alpha3 -> Alpha2, Some countries may use
+    private static String convertAlpha3ToAlpha2CountryCode(String countryCode) {
+        //todo: taking first two letters are not robust; is case ignored?
+        return ALPHA3_TO_ALPHA2_COUNTRY_CODE_MAP.getOrDefault(countryCode, countryCode.substring(0, 2));
+    }
+
+    private static Optional<String> countryFromMcc(String mcc) {
+        try {
+            return Optional.ofNullable(MCC_TO_COUNTRY_CODE_MAP.get(Integer.parseInt(mcc)));
+        } catch (NumberFormatException e) {
+            return Optional.empty();
+        }
+    }
+
+    private static Map<String, String> createAlpha3ToAlpha2CountryCodeMap() {
+        final Map<String, String> result = new HashMap<>();
+        result.put("AND", "AD");
+        result.put("AGO", "AO");
+        result.put("AUT", "AT");
+        result.put("BGD", "BD");
+        result.put("BLR", "BY");
+        result.put("CAF", "CF");
+        result.put("TCD", "TD");
+        result.put("CHL", "CL");
+        result.put("CHN", "CN");
+        result.put("COG", "CG");
+        result.put("COD", "CD");
+        result.put("DNK", "DK");
+        result.put("GNQ", "GQ");
+        result.put("EST", "EE");
+        result.put("GIN", "GN");
+        result.put("GNB", "GW");
+        result.put("GUY", "GY");
+        result.put("IRQ", "IQ");
+        result.put("IRL", "IE");
+        result.put("ISR", "IL");
+        result.put("KAZ", "KZ");
+        result.put("LBY", "LY");
+        result.put("MDG", "MG");
+        result.put("MDV", "MV");
+        result.put("MEX", "MX");
+        result.put("MNE", "ME");
+        result.put("MOZ", "MZ");
+        result.put("PAK", "PK");
+        result.put("PNG", "PG");
+        result.put("PRY", "PY");
+        result.put("POL", "PL");
+        result.put("PRT", "PT");
+        result.put("SRB", "RS");
+        result.put("SVK", "SK");
+        result.put("SVN", "SI");
+        result.put("SWE", "SE");
+        result.put("TUN", "TN");
+        result.put("TUR", "TR");
+        result.put("TKM", "TM");
+        result.put("UKR", "UA");
+        result.put("ARE", "AE");
+        result.put("URY", "UY");
```

Comment:
Maybe existant `CountryCodeMapper` class will do the job?

Reply:
will check, thanks